core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,358 @@
+        language_accent_code: str
+    ) -> voiceover_models.EntityVoiceoversModel:
+        """Returns a unique entity voiceovers model instance.
+
+        Args:
+            entity_type: str. The type of the entity.
+            entity_id: str. The ID of the entity.
+            entity_version: int. The version of the entity.
+            language_accent_code: str. The language-accent code of the
+                voiceover.
+
+        Returns:
+            EntityVoiceoversModel. An instance of the entity voiceover model.
+        """
+        with datastore_services.get_ndb_context():
+            entity_voiceovers_model = (
+                voiceover_models.EntityVoiceoversModel.create_new(
+                    entity_id, entity_type, entity_version,
+                    language_accent_code, {}
+                )
+            )
+        entity_voiceovers_model.update_timestamps()
+        return entity_voiceovers_model
+
+    @classmethod
+    def update_entity_voiceover_for_given_id(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        accent_code: str,
+        content_id: str,
+        voiceover_dict: state_domain.VoiceoverDict,
+        entity_voiceover_id_to_entity_voiceovers: Dict[
+            str, voiceover_models.EntityVoiceoversModel]
+    ) -> Dict[str, voiceover_models.EntityVoiceoversModel]:
+        """Updates the entity voiceover model instance for the given language
+        accent code for a specific content of an exploration.
+
+        Args:
+            entity_type: str. The entity type of an entity voiceovers instance.
+            entity_id: str. The given entity ID.
+            entity_version: int. The version number of the given exploration.
+            accent_code: str. The language accent code for the given entity
+                voiceover instance.
+            content_id: str. The content ID for a specific content in an
+                exploration.
+            voiceover_dict: VoiceoverDict. A voiceover dict that should be
+                added to the entity voiceover instance of the given language
+                accent code of the content.
+            entity_voiceover_id_to_entity_voiceovers:
+                dict(str, EntityVoiceoversModel). A dict with entity voiceover
+                ID as keys and its corresponding EntityVoiceoversModel as
+                values.
+
+        Returns:
+            dict(str, EntityVoiceoversModel). The updated dict with entity
+            voiceover ID as keys and its corresponding EntityVoiceoversModel as
+            values.
+        """
+        entity_voiceover_id = (
+            voiceover_models.EntityVoiceoversModel.generate_id(
+                entity_type,
+                entity_id,
+                entity_version,
+                accent_code
+            )
+        )
+
+        if entity_voiceover_id not in entity_voiceover_id_to_entity_voiceovers:
+            entity_voiceovers_object = (
```

Comment:
Do not confuse domain objects and models -- the method is named "create model instance" but you are naming the variable as an object.

Generally speaking, you should only deal with objects here and only handle model-layer interactions at the start and end of the job.